teamsyncd/teamsync.cpp

```diff
@@ -131,42 +132,59 @@ TeamSync::TeamPortSync::~TeamPortSync()
     }
 }
 
-int TeamSync::TeamPortSync::onPortChange(bool isInit)
+int TeamSync::TeamPortSync::onOptionChange()
 {
     struct team_port *port;
+    set<string> tmp_lag_members;
+
+    /* Check each port  */
     team_for_each_port(port, m_team)
     {
-        if (isInit || team_is_port_changed(port))
+        uint32_t ifindex;
+        char ifname[MAX_IFNAME + 1] = {0};
+        bool enabled;
+
+        ifindex = team_get_port_ifindex(port);
+        team_ifindex2ifname(m_team, ifindex, ifname, MAX_IFNAME);
+        team_get_port_enabled(m_team, ifindex, &enabled);
+
+        if (enabled)
+        {
+            tmp_lag_members.insert(string(ifname));
+        }
+    }
+
+    /* Compare old and new LAG members and set/del accordingly */
+    for (auto member : tmp_lag_members)
+    {
+        if (m_lagMembers.find(member) == m_lagMembers.end())
         {
-            string key = m_lagName;
-            key += ":";
-            uint32_t ifindex = team_get_port_ifindex(port);
-            char ifname[MAX_IFNAME + 1] = {0};
-            key += team_ifindex2ifname(m_team, ifindex, ifname, MAX_IFNAME);
-
-            if (team_is_port_removed(port))
-            {
-                m_lagTable->del(key);
-            } else
-            {
-                std::vector<FieldValueTuple> fvVector;
-                FieldValueTuple l("linkup", team_is_port_link_up(port) ? "up" : "down");
-                FieldValueTuple s("speed", to_string(team_get_port_speed(port)) + "Mbit");
-                FieldValueTuple d("duplex", team_get_port_duplex(port) ? "full" : "half");
-                fvVector.push_back(l);
-                fvVector.push_back(s);
-                fvVector.push_back(d);
-                m_lagTable->set(key, fvVector);
-            }
+            string key = m_lagName + ":" + member;
+            vector<FieldValueTuple> v;
+            FieldValueTuple l("active_status", "up");
+            v.push_back(l);
+            m_lagTable->set(key, v);
         }
     }
+
+    for (auto member : m_lagMembers)
+    {
+        if (tmp_lag_members.find(member) == tmp_lag_members.end())
+        {
+            string key = m_lagName + ":" + member;
+            m_lagTable->del(key);
```

Comment:
I feel the app db lag contract should be lag member enable/disable, not add/remove.

when a lag member is disabled, we should still have the entry in the app db, but the active_status is down.

can you also update the schema doc?